packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -205,38 +147,33 @@ html<TableColumnMappingHeaderView>`${x => x.getMappingToRender().groupHeaderView
 
 ```HTML
 <${this.icon}
-    title="${x => x.label}"
-    aria-label="${x => x.label}"
+    title="${x => x.labelHidden ? x.label: null}"
+    aria-label="${x  => x.labelHidden ? x.label: null}"
+    aria-hidden="${x => x.labelHidden ? 'false' : 'true'}"
     severity="${x => x.severity}">
 </${this.icon}>
+${when(() => !x.labelHidden, html`
+    <span>${x => x.label}</span>
+`)}
 ```
 
 `mapping.groupHeaderViewTemplate`:
 
 ```HTML
 <${this.icon}
-    title="${x => x.label}"
-    aria-label="${x => x.label}"
+    aria-hidden="${x => x.labelHidden ? 'false' : 'true'}"
     severity="${x => x.severity}">
 </${this.icon}>
-<span
-    ${ref('span')}
-    @mouseover="${x => {
-        x.isValidContentAndHasOverflow = !!x.label && x.span!.offsetWidth < x.span!.scrollWidth;
-    }}"
-    @mouseout="${x => {
-        x.isValidContentAndHasOverflow = false;
-    }}"
-    title=${x => (x.isValidContentAndHasOverflow ? x.label : null)}>
+<span>
```

Comment:
Note: I removed the `mouseover`/`mouseout` code from here because I didn't think it was particularly relevant to the plan here. We now have the `overflow` behavior that we'll use, and the HLD already says that we'll show the title if the label gets ellipsized.